File: CPP/26. Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() == 1){
            return 1;
        }
        
        int leftPointer = 1;
        for(int i = 1 ; i< nums.size() ; i++){
            if(nums[i] > nums[leftPointer-1]){
                if(i != leftPointer){
                    swap(nums[i], nums[leftPointer]);
                }
                leftPointer++;
            }
        }
        
        return leftPointer;
    }
};

File: CSharp/26. Remove Duplicates from Sorted Array.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        int leftPointer = 1;
        if(nums.Length == 1){
            return 1;
        }
        
        
        for(int i = 1 ; i< nums.Length ; i++){
            if(nums[leftPointer-1] < nums[i]){
                if(i != leftPointer){
                    swap(ref nums[i] , ref nums[leftPointer]);
                }
                leftPointer++;
            }
        }
        
        return leftPointer;
    }
    
    private void swap(ref int a, ref int b){
        int temp = a;
        a = b;
        b = temp;
    }
}